catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStorePublisher.java

```diff
@@ -0,0 +1,201 @@
+        this.federationAdminService = federationAdminService;
+    }
+
+    public void setRegistryPublicationService(
+            RegistryPublicationService registryPublicationService) {
+        this.registryPublicationService = registryPublicationService;
+    }
+
+    public void bindRegistryStore(ServiceReference reference) {
+        BundleContext bundleContext = getBundleContext();
+
+        if (reference == null || bundleContext == null || bundleContext.getService(reference) == null) {
+            return;
+        }
+
+        RegistryStore registryStore = (RegistryStore) bundleContext.getService(reference);
+
+        Boolean previousState = registryStoreMap.put(registryStore.getConfigurationPid(),
+                registryStore.isAutoPush());
+
+        SourceMonitor storeRegisteredSourceMonitor = new SourceMonitor() {
+            @Override
+            public void setAvailable() {
+                if (registryStore.isAutoPush()) {
+                    registryPublish(registryStore);
+                }
+            }
+
+            @Override
+            public void setUnavailable() {
+            }
+        };
+
+        if (registryStore.isAvailable(storeRegisteredSourceMonitor) && registryStore.isAutoPush()) {
+            if (previousState == null || !previousState) {
+                registryPublish(registryStore);
+            }
+        }
+    }
+
+    public void unbindRegistryStore(ServiceReference reference) {
+
+        BundleContext bundleContext = getBundleContext();
+
+        if (reference != null && bundleContext != null) {
+            RegistryStore registryStore = (RegistryStore) bundleContext.getService(reference);
+
+            registryStoreMap.remove(registryStore.getConfigurationPid());
+        }
+    }
+
+    void registryPublish(RegistryStore registryStore) {
+
+        if (registryStore.getRegistryId()
+                .isEmpty()) {
+            return;
+        }
+        try {
+
+            Optional<Metacard> registryIdentityMetacardOpt =
+                    Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
+
+            if (registryIdentityMetacardOpt.isPresent()) {
+                Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
+                String localRegistryId = registryIdentityMetacard.getAttribute(
+                        RegistryObjectMetacardType.REGISTRY_ID)
+                        .getValue()
+                        .toString();
+                Security.runAsAdminWithException(() -> {
+                    registryPublicationService.publish(localRegistryId,
+                            registryStore.getRegistryId());
+                    return null;
+                });
+            }
+        } catch (Exception e) {
+            LOGGER.error("Failed to publish registry configuration to {}",
+                    ((RegistryStoreImpl) registryStore).getRemoteName());
+        }
+    }
+
```

Comment:
Agreed. The only differences are the lambda passed through to security to run and the "unpublish" or "publish" of your error.